### DISCUSSION TITLE
Simulator compatibility

### DISCUSSION THREAD

**Discussion by niclashoyer** [joeycastillo/Sensor-Watch#107] [Q&A]
Is there a list of possible differences / caveats when using the simulator?

I'm currently investigating a nasty problem while hacking on #106 where I thought this might be a deep sleep issue. But then I found out that `EVENT_LOW_ENERGY_UPDATE` is never fired. My current guess is, that the simulator might not work properly when not in focus (background tab in browser).

I think that the simulation is stopped or runs much slower when the tab is not in focus. At least the `EVENT_TICK` is not fired properly and the timestamps are off. It also misses [this comparison](https://github.com/joeycastillo/Sensor-Watch/blob/main/movement/movement.c#L158) in the check for scheduled tasks, so that the `EVENT_BACKGROUND_TASK` is never fired.

If the simulator stays on screen / in focus all the time, this does not happen.

**Comment by yasuiniko** [comment]
Even when the simulator is in focus, it becomes unresponsive for me after `le_mode_ticks` runs out (roughly speaking; I have never debugged C before and have not attempted it yet). I have only tested with 5 seconds of `movement_le_inactivity_deadlines` and `alarm_time.unit.second = 3`. I tried safari and firefox, with the same behaviour.
